src/bd_rhapsody/bd_rhapsody_sequence_analysis/_process_cwl.R

```diff
@@ -0,0 +1,115 @@
+library(tidyverse)
+
+# fetch and read cwl file
+lines <- read_lines("https://bitbucket.org/CRSwDev/cwl/raw/8feeace1141b24749ea6003f8e6ad6d3ad5232de/v2.2.1/rhapsody_pipeline_2.2.1.cwl")
+cwl_header <- lines[[1]]
+cwl_obj <- jsonlite::fromJSON(lines[-1], simplifyVector = FALSE)
+
+# recursively remove objects with `"class": "DockerRequirement"`
+remove_docker <- function(x) {
+  if (is.list(x)) {
+    if ("class" %in% names(x) && x$class == "DockerRequirement") {
+      return(NULL)
+    } else {
+      out <- map(x, remove_docker)
+      out2 <- out[!sapply(out, is.null)]
+      return(out2)
+    }
+  } else {
+    return(x)
+  }
+}
+cwl_obj_2 <- remove_docker(cwl_obj)
```

Comment:
you can indeed run `cwl-runner --no-container ...` but it used to be the case that the different steps had the Docker container marked as required and so the pipeline would fail anyway. Hang on, let me try again.

Reply:
Seems to be fixed now! I see that the DockerRequirements have been moved to hints as opposed to strict requirements :relaxed:

Reply:
Wohoo! that should clean things up 👍